### ISSUE TITLE
MD018: False positive on Rust attributes inside code blocks

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#274]
## Summary

MD018 (no-missing-space-atx) incorrectly flags Rust attributes like `#[no_mangle]` inside code blocks as headings without space after hash.

## Example

```markdown
'''
#[unsafe(no_mangle)]
pub extern "C" fn call_from_c() {
    println!("Just called a Rust function from C!");
}
'''
```

We flag `#[unsafe(no_mangle)]` with:
```
MD018/no-missing-space-atx: No space after hash on atx style heading
```

## Root Cause

The rule is checking for `#` at start of line but not respecting code block boundaries.

## Impact

| Linter | MD018 Violations |
|--------|------------------|
| mdbook-lint | 5 |
| rumdl | 0 |
| markdownlint | 0 |

All 5 of our violations are false positives from Rust attributes in code blocks.

## Related

Issue #263 - Linter comparison

**Comment by joshrotenberg** [comment]
Fixed in PR #279.
